app/src/sharedTest/java/org/oppia/android/app/databinding/MarginBindableAdaptersTest.kt

```diff
@@ -0,0 +1,363 @@
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.exploration.lightweightcheckpointing.ExplorationStorageModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.platformparameter.PlatformParameterModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.TestImageLoaderModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [MarginBindableAdapters]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = MarginBindableAdaptersTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class MarginBindableAdaptersTest {
+
+  @Inject
+  lateinit var context: Context
+
+  @get:Rule
+  var activityRule: ActivityScenarioRule<MarginBindableAdaptersTestActivity> =
+    ActivityScenarioRule(
+      Intent(
+        ApplicationProvider.getApplicationContext(),
+        MarginBindableAdaptersTestActivity::class.java
+      )
+    )
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    Intents.init()
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  @Config(qualifiers = "port")
+  @Test
+  fun testMarginBindableAdapters_ltrIsEnabled_port_marginStartAndMarginEndForLtrIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_textview)
+      setLayoutMarginStart(
+        textView,
+        context.resources.getDimension(R.dimen.content_item_exploration_view_margin_start)
+      )
+      setLayoutMarginEnd(
+        textView,
+        context.resources.getDimension(R.dimen.content_item_exploration_view_margin_end)
+      )
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginStart.toFloat()).isEqualTo(
```

Comment:
We should either convert to integers, or use isWithin for proper float comparison:

```kotlin
assertThat(floatSubject).isWithin(TOLERANCE).of(expectedValue)
```

Where TOLERANCE is something like:

```kotlin
private const val TOLERANCE = 1e-5f
```

Reply:
Used `assertThat(floatSubject).isWithin(TOLERANCE).of(expectedValue)` for float comparison.